pkg/server/auth/auth.go

```diff
@@ -66,6 +69,7 @@ func Principal(ctx context.Context) *UserPrincipal {
 type UserPrincipal struct {
 	ID     string   `json:"id"`
 	Groups []string `json:"groups"`
+	Token  string   `json:"-"`
```

Comment:
Small reservation here, wrt bundling the token into the Principal. The `-` json tricks works great for responses, but I guess the principal might be quite easy to accidentally include in logging etc, where `%s` seems to still include it.

Maybe there are other tricks? Like implementing `String()`?

Reply:
Ah yes, right you are. I will add a `String()` method to the `UserPrincipal` struct to avoid this.

Reply:
Fixed!